app/pulp/app/viewsets/repository.py

```diff
@@ -72,6 +72,15 @@ class ImporterViewSet(NamedModelViewSet):
     serializer_class = ImporterSerializer
     endpoint_name = 'importers'
 
+    def destroy(self, request, name):
+        imp = self.get_object()
+        async_result = importer.delete.apply_async_with_reservation(
+            tags.RESOURCE_REPOSITORY_IMPORTER_TYPE, imp.name,
+            kwargs={'repo_name': imp.repository,
```

Comment:
If we'd like to specify repository name, isn't it supposed to be `imp.repository.name` and not just `imp.repository` ? It looks more explicit to me that way.